### PR TITLE
Neodata - Corrected issues

### PR DESCRIPTION
**Pull request by neodatagroup**

### CODE REVIEW

**Comment by neodatagroup** [comment]
Refers to https://github.com/ampproject/amphtml/pull/3914

**Comment by lannka** [comment]
@neodatagroup I saw you had 2 PR for the same things. Could you consolidate? If one of them is abandoned, please close it.

**Comment by neodatagroup** [comment]
@lannka Just closed the other one.
Thanks

File: ads/_config.js [hunk 1]
```diff
@@ -176,6 +179,21 @@ export const adPreconnect = {
     'https://mads.at.atwola.com',
     'https://aka-cdn.adtechus.com',
   ],
+  neo4w: [
+    'https://js-ssl.neodatagroup.com',
```
Comment (lannka) [inline ads/_config.js]: the hosts already in prefetch not needed here.

File: ads/_config.js [hunk 2]
```diff
@@ -176,6 +179,21 @@ export const adPreconnect = {
...
+    'https://ad-aws-it.neodatagroup.com',
+    'https://tracker.neodatagroup.com'
+  ],
+  neo4wPPN: [
+    'https://js-ssl.neodatagroup.com',
```
Comment (lannka) [inline ads/_config.js]: the hosts already in prefetch not needed here.

File: ads/_config.js [hunk 3]
```diff
@@ -176,6 +179,21 @@ export const adPreconnect = {
...
+    'https://ad-aws-it.neodatagroup.com',
+    'https://tracker.neodatagroup.com'
+  ],
+  adagio: [
+    'https://js-ssl.neodatagroup.com',
```
Comment (lannka) [inline ads/_config.js]: the hosts already in prefetch not needed here.

File: examples/ads.amp.html [hunk 4]
```diff
@@ -40,6 +40,28 @@
...
+          data-pid="ansa_ansa_tecnologia">
+  </amp-ad>
+
+  <h2>4W PPN</h2>
+  <amp-ad width=320 height=50
+          type="4wppn"
+          data-loc="ppnm_ans_ans_ros_top">
+  </amp-ad>
+
+  <h2>Ad.Agio</h2>
```
Comment (lannka) [inline examples/ads.amp.html]: is this a typo?

File: ads/adagio.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Please refer to [Neodata Help Desk](mailto:adagio@neodatagroup.com) for more
+information on how to get required parameters.
+
+## Examples
+
+### Simple ad tag 
+
+```html
+ <amp-ad width=320 height=250
+          type="adagio"
```
Comment (lannka) [inline ads/adagio.md]: please fix indentation

File: examples/ads.with.script.amp.html [hunk 6]
```diff
@@ -41,6 +41,19 @@
...
+          data-taxonomy="/ansa/ansa/ansa.it/tecnologia"
+          data-pid="ansa_ansa_tecnologia">
+  </amp-ad>
+
+  <h2>4W PPN</h2>
+  <amp-ad width=320 height=50
+          type="4wppn"
+          data-loc="ppnm_ans_ans_ros_top">
+  </amp-ad>
+
```
Comment (lannka) [inline examples/ads.with.script.amp.html]: please also add adagio

**Comment by lannka** [comment]
Please also resolve the merge conflicts. Otherwise LGTM.

File: examples/ads.amp.html (hunk 4 above)
Comment (neodatagroup) [inline examples/ads.amp.html]: Sorry I'm not getting the point, the two examples refer to two different types

File: examples/ads.with.script.amp.html (hunk 6 above)
Comment (neodatagroup) [inline examples/ads.with.script.amp.html]: ad.agio is present few lines below

File: ads/_config.js (hunk 2 above)
Comment (neodatagroup) [inline ads/_config.js]: @lannka  so we can remove from there ? We need just one for all the three ?

**Comment by neodatagroup** [comment]
Hi, 
just pushed change that should fix all open issues,
please confirm if something still is missing
Thanks

**Comment by lannka** [comment]
@neodatagroup pls resolve merge conflicts

**Comment by neodatagroup** [comment]
@lannka 
sorry I'm not too familiar with Git, looking at repo I cannot find conflicts
MacBook-Pro-di-Salvo-2:amphtml nics$ git status
On branch neodatagroup2
Your branch is up-to-date with 'origin/neodatagroup2'.
nothing to commit, working directory clean

Can  you help me ?

**Comment by kul3r4** [comment]
Hi,
you are working on a branch in a fork, so you need to get the latest from the amphtml repository.

If you haven't already setup the upstream, try: 
`git remote add upstream https://github.com/ampproject/amphtml.git`
then
```
git fetch upstream
git rebase upstream/master
```

This will get the latest changes from the amphtml main repository and apply to your fork. From here, you should be able to solve the conflict, add the files marked as changed and conclude the rebase.

**Comment by neodatagroup** [comment]
@kul3r4 
Should be all fixed

**Comment by kul3r4** [comment]
Thanks,
have a look at the errors from Travis and please sign the CLA.

**Comment by neodatagroup** [comment]
I signed it!

**Comment by neodatagroup** [comment]
I've corrected the email address in the CLA in order to fix the issue, 
can you confirm that is correct now ?
Thanks

**Comment by lannka** [comment]
@neodatagroup did you make any commit after changing the email address in git?
the googlerobot will auto approve if the check went through.

File: ads/adagio.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+// adagio js
+import {loadScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adagio(global, data) {
+
+  //checkData(data, ['sid', 'loc']);
```
Comment (lannka) [inline ads/adagio.js]: please sync to the head of master branch. we switched the `checkData` util to `validateData`

File: examples/ads.amp.html [hunk 8]
```diff
@@ -129,6 +132,15 @@ <h2 id="a9">A9</h2>
       data-aax_src="302">
   </amp-ad>
 
+  <h2 id="adagio">Ad.Agio</h2>
+  <amp-ad width=320 height=250
+          type="adagio"
```
Comment (lannka) [inline examples/ads.amp.html]: Please follow the code style and match the indentation with other examples.

File: ads/4wppn.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  const $neodata = global;
+
+  $neodata._neodata = {};
+  $neodata._neodata.adagioTag = [];
+  $neodata._neodata.amp = data;
+
+  $neodata._neodata.adagioTag.push({
+    'loc': data.loc,
+    'ph': 'c',
+  });
```
Comment (lannka) [inline ads/4wppn.js]: you can probably do:

```
global._neodata = {
  adagioTag: [{
    'loc': data.loc,
    'ph': 'c',
  }],
  amp: data,
}

loadScript(global, 'https://js-ssl.neodatagroup.com/uploads/js/4wm.js');
```

Comment (neodatagroup) [inline ads/4wppn.js]: Done, thanks

File: examples/ads.amp.html (hunk 8 above)
Comment (neodatagroup) [inline examples/ads.amp.html]: Done

File: ads/adagio.js (hunk 7 above)
Comment (neodatagroup) [inline ads/adagio.js]: Done

**Comment by lannka** [comment]
@neodatagroup you got 108 files in the PR :-).
To fix, please sync your forked master branch to upstream master.

Something like:
```
git checkout master
git pull upstream master
git push origin master
```

**Comment by kul3r4** [comment]
@neodatagroup Hi, have you managed to pull the changes from master? Lt us know if you need any help.

**Comment by neodatagroup** [comment]
@kul3r4 unfortunately no, I've tried your suggestion but no luck.
Could we have a 10 min hangout in screen share to review it together ?
Thanks

**Comment by kul3r4** [comment]
@neodatagroup Could you take a look at the failing check?
AssertionError: Missing config for [4w]: expected { Object (_ping_, a9, ...) } to contain key '4w'
	    at /tmp/b38dd10b9266bec45b101a299ea3b896.browserify:95230:86 <- /home/travis/build/ampproject/amphtml/test/functional/test-ads-config.js:23:68
	    at Array.forEach (native)
	    at Context.<anonymous> (/tmp/b38dd10b9266bec45b101a299ea3b896.browserify:95229:41 <- /home/travis/build/ampproject/amphtml/test/functional/test-ads-config.js:22:40)

File: ads/_config.js [hunk 10]
```diff
@@ -40,6 +40,10 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  4w: {},
```
Comment (jridgewell) [inline ads/_config.js]: Keys cannot start with a numeral. You'll have to surround them in quotes:

```js
  },
  '4w': {},
  '4wppn': {},
  //...

**Comment by lannka** [comment]
Closing this for now. Feel free to reopen if you're still working on it.

**Pull request closed without merging**